docs/docs/classic-ml/getting-started/hyperparameter-tuning/index.mdx

```diff
@@ -1,590 +1,159 @@
-
-1. **Switch to Chart view**: Click the "Chart" tab
-2. **Create parallel coordinates plot**:
-   - Select "Parallel coordinates"
-   - Add `learning_rate` and `momentum` as coordinates
-   - Set `val_rmse` as the metric
-3. **Interpret the visualization**:
-   - Blue lines = better performing runs
-   - Red lines = worse performing runs
-   - Look for patterns in successful parameter combinations
-
-### Key Insights to Look For
-
-- **Learning rate patterns**: Too high causes instability, too low causes slow convergence
-- **Momentum effects**: Moderate momentum (0.3-0.7) often works best
-- **Training curves**: Check artifacts to see if models converged properly
-
-## Step 6: Register Your Best Model
+## Step 3: Define the objective function
 
-Time to promote your best model to production:
+In Optuna, a **study** is a single optimization task, representing the entire hyperparameter tuning
+session consisting of multiple **trials**. A trial is a single execution of the objective function,
+namely, training a model with a single combination of hyperparameters.
 
-1. **Find the best run**: In the Table view, click on the run with the lowest `val_rmse`
-2. **Navigate to model artifacts**: Scroll to the "Artifacts" section
-3. **Register the model**:
-   - Click "Register Model" next to the model folder
-   - Enter model name: `wine-quality-predictor`
-   - Add description: "Optimized neural network for wine quality prediction"
-   - Click "Register"
+In MLflow, this structure is represented by a **parent run** and **child runs**. A parent run is a
+run that contains all the child runs for different trials. The parent-child relationship allows us
+to keep track of each trial during the hyperparameter tuning process and group them nicely in the
+MLflow UI.
 
-4. **Manage model lifecycle**:
-   - Go to "Models" tab in MLflow UI
-   - Click on your registered model
-   - Transition to "Staging" stage for testing
-   - Add tags and descriptions as needed
+First, let's define the objective function that is executed for each trial. To log the parameters,
+metrics, and model file, we use MLflow's API inside the objective function. An MLflow run is created
+with the `nested=True` flag to indicate it is a child run.
 
-## Step 7: Deploy Your Model Locally
-
-Test your model with a REST API deployment:
-
-```bash
-# Serve the model (choose the version number you registered)
-mlflow models serve -m "models:/wine-quality-predictor/1" --port 5002
-```
+```python
+import mlflow
+import sklearn
 
-:::note Port Configuration
-We use port 5002 to avoid conflicts with the MLflow UI running on port 5000. In production, you'd typically use port 80 or 443.
-:::
 
-### Test Your Deployment
+def objective(trial):
+    # Setting nested=True will create a child run under the parent run.
+    with mlflow.start_run(nested=True, run_name=f"trial_{trial.number}"):
+        rf_max_depth = trial.suggest_int("rf_max_depth", 2, 32)
+        params = {"max_depth": rf_max_depth}
```

Comment:
We might want to do 3 or more parameters here in order to leverage all of the visualization (parallel coordinates plot) and to show a slightly more realistic example.